src/pseudo_intrinsics.h

```diff
@@ -34,8 +34,57 @@
 #include "stdint.h"
 #include "common.h" /* for is_aligned() */
 
-/*************************** AltiVec (Power) ***************************/
-#ifdef __ALTIVEC__
+/*************************** NEON (ARM) *******************************/
+#ifdef __ARM_NEON__
+#include <arm_neon.h>
+
+typedef uint32x4_t vtype32;
+typedef uint64x2_t vtype64;
+typedef vtype64 vtype;
+
+#define vadd_epi32              vaddq_u32
+#define vadd_epi64              vaddq_u64
+#define vand                    vandq_u32
```

Comment:
This is troublesome, just like in AltiVec. Although `vandq_u32` and `vandq_u64` translate to the same instruction, they have different function signatures, thus are not interchangeable.

I'm thinking maybe we should have different versions of `vand`, `vload`, etc, which would make it more natural to write AltiVec and NEON intrinsics. So far only x86 uses the type-agnostic form of those intrinsics. 

We could also distinguish between `vtype32` and `vtype64`, which may improve readability (knowing what exactly type of data we are dealing with).